### PR TITLE
Document rotation of IAM user access keys

### PR DESCRIPTION
**Pull request by rogeruiz**
Part of secret rotation, this documentation covers versioning the access_key and secret_access_key during a secret rotation.

### CODE REVIEW

**Comment by LinuxBozo** [comment]
I have a concern that without proper diff context, all these examples will lose a bit of meaning. Very hard to see from the examples what exactly is going on if you really aren't paying close attention.

**Comment by rogeruiz** [comment]
Yeah very much agree. Maybe having more generic examples _with diffs_  makes more sense. @LinuxBozo 

The `iam_user/base` and `iam_user/rotation` will be two that make sense to include. But if we make the actual modules to update more generic we could even point to this as a start for creating `iam_user` modules. 🤷‍♂️

**Comment by rogeruiz** [comment]
Noticed some typos. So this isn't ready to merge yet.

**Comment by rogeruiz** [comment]
This should now be ready to be reviewed. @jmcarp @rememberlenny

File: content/docs/ops/rotating-iam-users.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
++  user = "${aws_iam_user.iam_user.name}"
++}
++
+ resource "aws_iam_user_policy" "iam_policy" {
+   name = "${aws_iam_user.iam_user.name}-policy"
+   user = "${aws_iam_user.iam_user.name}"
+
+```
+
+The example above leverages `_v2` as the now current version by removing the
```
Comment (cnelson) [inline content/docs/ops/rotating-iam-users.md]: Should this be v2 / v3, not v1/v2 (based on your sample diff)

File: content/docs/ops/rotating-iam-users.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
++  user = "${aws_iam_user.iam_user.name}"
++}
++
+ resource "aws_iam_user_policy" "iam_policy" {
+   name = "${aws_iam_user.iam_user.name}-policy"
+   user = "${aws_iam_user.iam_user.name}"
+
+```
+
+The example above shows keys being rotated from `_v1` to `_v2`. The example
```
Comment (cnelson) [inline content/docs/ops/rotating-iam-users.md]: Should this be v2 / v3, not v1/v2 (based on your sample diff)

File: content/docs/ops/rotating-iam-users.md (hunk 1 above)
Comment (rogeruiz) [reply content/docs/ops/rotating-iam-users.md]: good catch. thanks.

**Comment by rogeruiz** [comment]
completely forgot to push up my changes. Should be resolved now @cnelson

**Review by cnelson — COMMENTED** [review]
LGTM me, but I think @LinuxBozo should approve since he raised the initial issue.  Does this make more sense with the additional context?

**Comment by rogeruiz** [comment]
True. I just wish we had a syntax highlighter for the diffs. 😢

File: content/docs/ops/rotating-iam-users.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+ output "access_key_id_prev" {
+-  value = ""
++  value = "${aws_iam_access_key.iam_access_key_v2.access_key_id}"
+ }
+ output "secret_access_key_prev" {
+-  value = ""
++  value = "${aws_iam_access_key.iam_access_key_v2.secret_access_key}"
+ }
+ output "access_key_id_curr" {
+-  value = "${aws_iam_access_key.iam_access_key.access_key_id}"
```
Comment (LinuxBozo) [inline content/docs/ops/rotating-iam-users.md]: Shouldn't this old value be v2?

File: content/docs/ops/rotating-iam-users.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
++  value = "${aws_iam_access_key.iam_access_key_v3.access_key_id}"
+ }
+ output "secret_access_key_curr" {
+-  value = "${aws_iam_access_key.iam_access_key.secret_access_key}"
```
Comment (LinuxBozo) [inline content/docs/ops/rotating-iam-users.md]: This old value should be v2 as well, yeah?

**Review by LinuxBozo — CHANGES_REQUESTED** [review]

**Review by jmcarp — APPROVED** [review]

**Review by LinuxBozo — APPROVED** [review]

**Pull request merged**